Source/Core/Core/NetPlayClient.cpp

```diff
@@ -973,70 +972,36 @@ bool NetPlayClient::WiimoteUpdate(int _number, u8* data, const u8 size)
   {
     std::lock_guard<std::recursive_mutex> lkp(m_crit.players);
 
-    // in game mapping for this local Wiimote
-    unsigned int in_game_num = LocalWiimoteToInGameWiimote(_number);
-    // does this local Wiimote map in game?
-    if (in_game_num < 4)
+    // Only send data, if this Wiimote is mapped to this player
+    if (m_wiimote_map[_number] == m_local_player->pid)
     {
-      if (m_wiimote_current_data_size[in_game_num] == size)
+      nw.assign(data, data + size);
+      do
       {
-        nw.assign(data, data + size);
-        do
-        {
-          // add to buffer
-          m_wiimote_buffer[in_game_num].Push(nw);
+        // add to buffer
+        m_wiimote_buffer[_number].Push(nw);
 
-          SendWiimoteState(in_game_num, nw);
-        } while (m_wiimote_buffer[in_game_num].Size() <=
-                 m_target_buffer_size * 200 /
-                     120);  // TODO: add a seperate setting for wiimote buffer?
-      }
-      else
-      {
-        while (m_wiimote_buffer[in_game_num].Size() > 0)
-        {
-          // Reporting mode changed, so previous buffer is no good.
-          m_wiimote_buffer[in_game_num].Pop();
-        }
-        nw.resize(size, 0);
-
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_buffer[in_game_num].Push(nw);
-        m_wiimote_current_data_size[in_game_num] = size;
-      }
+        SendWiimoteState(_number, nw);
+      } while (m_wiimote_buffer[_number].Size() <=
+               m_target_buffer_size * 200 /
+                   120);  // TODO: add a seperate setting for wiimote buffer?
     }
 
   }  // unlock players
 
-  while (m_wiimote_current_data_size[_number] == size && !m_wiimote_buffer[_number].Pop(nw))
+  while (!m_wiimote_buffer[_number].Pop(nw))
   {
     // wait for receiving thread to push some data
     Common::SleepCurrentThread(1);
```

Comment:
As a note, it would likely be way better for performance to make this a Common::Event or (maybe even better) restructure around Common::BlockingLoop.